### PR TITLE
Add smart category toggling

### LINKED ISSUE #1: Smart category toggling
If toggling a category would result in no recipes being shown, then that category should be visually disabled. It should also not respond to user clicks in this state.

### PR DESCRIPTION
**Pull request by diggingforfire**
First implementation of smart category toggling (html needs to be refactored at some point)

### CODE REVIEW

File: src/app/recipes/recipes.component.html [hunk 1]
```diff
@@ -3,31 +3,31 @@
     <div><fa-icon [icon]="faUtensils" class="text-3xl"></fa-icon></div>
     <div>Alle recepten</div>
   </li>
-  <li (click)="filter(3)" [ngClass]="{ 'bg-amber-300': filteredCategories.includes(3)  }" class="items-center text-center bg-amber-100 p-2 cursor-pointer">
+  <li (click)="filter(3)" [ngClass]="{ 'bg-amber-300': filteredCategories.includes(3), 'opacity-25 pointer-events-none': getRecipeCount(3) == 0 }" class="items-center text-center bg-amber-100 p-2 cursor-pointer">
```
Comment (LauwiMeara) [inline src/app/recipes/recipes.component.html]: To keep logic out of the template, let getRecipeCount() return a boolean (e.g. 'hasNoRecipes').

File: src/app/recipes/recipes.component.ts [hunk 2]
```diff
@@ -42,7 +42,17 @@ export class RecipesComponent {
...
   }
 
-  getFilteredRecipes(): Recipe[] {
-    return this.recipes.filter(recipe => this.filteredCategories.every (id => recipe.categories.some(category => category.id == id)));
+  getRecipeCount(categoryId: number): number {
+    return this.getFilteredRecipes(categoryId).length;
+  }
+
+  getFilteredRecipes(categoryId: number | null = null): Recipe[] {
+    let filter = this.filteredCategories.slice();
```
Comment (LauwiMeara) [inline src/app/recipes/recipes.component.ts]: Can this name be changed? It is confusing because of the existing function filter() in RecipesComponent.

**Review by LauwiMeara — CHANGES_REQUESTED** [review]
Cool feature!

File: src/app/recipes/recipes.component.html (hunk 1 above)
Comment (diggingforfire) [reply src/app/recipes/recipes.component.html]: Done!

File: src/app/recipes/recipes.component.ts (hunk 2 above)
Comment (diggingforfire) [reply src/app/recipes/recipes.component.ts]: Done

**Review by LauwiMeara — APPROVED** [review]
Looks good!

**Pull request merged**